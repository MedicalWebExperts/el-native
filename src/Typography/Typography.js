import React from 'react';
import { object, bool, string, number } from 'prop-types';
import { Text } from 'react-native';

import Theme from '../Theme';

const theme = Theme.getTheme();
const styles = theme.typography;

const h1Proptypes = {
  /**
   * The prop "style" overrides the custom styles set in the component.
   */
  style: object,
  /**
   * The prop "color" defines the color of the text.
   */
  color: bool,
  /**
   * The prop "numberOfLines" receives a number which limits the lines shown
   * (an ellipsis will show at the end when truncated)
   */
  numberOfLines: number,
  /**
   * The prop "children" defines the content that will live inside the component.
   */
  children: string.isRequired,
};
const h1DefaultProps = {
  style: {},
  color: false,
  numberOfLines: null,
};
const h2Proptypes = {
  /**
   * The prop "style" overrides the custom styles set in the component.
   */
  style: object,
  /**
   * The prop "color" defines the color of the text.
   */
  color: bool,
  /**
   * The prop "numberOfLines" receives a number which limits the lines shown
   * (an ellipsis will show at the end when truncated)
   */
  numberOfLines: number,
  /**
   * The prop "children" defines the content that will live inside the component.
   */
  children: string.isRequired,
};
const h2DefaultProps = {
  style: {},
  color: false,
  numberOfLines: null,
};
const h3Proptypes = {
  /**
   * The prop "style" overrides the custom styles set in the component.
   */
  style: object,
  /**
   * The prop "color" defines the color of the text.
   */
  color: bool,
  /**
   * The prop "numberOfLines" receives a number which limits the lines shown
   * (an ellipsis will show at the end when truncated)
   */
  numberOfLines: number,
  /**
   * The prop "children" defines the content that will live inside the component.
   */
  children: string.isRequired,
};
const h3DefaultProps = {
  style: {},
  color: false,
  numberOfLines: null,
};
const h4Proptypes = {
  /**
   * The prop "style" overrides the custom styles set in the component.
   */
  style: object,
  /**
   * The prop "color" defines the color of the text.
   */
  color: bool,
  /**
   * The prop "numberOfLines" receives a number which limits the lines shown
   * (an ellipsis will show at the end when truncated)
   */
  numberOfLines: number,
  /**
   * The prop "children" defines the content that will live inside the component.
   */
  children: string.isRequired,
};
const h4DefaultProps = {
  style: {},
  color: false,
  numberOfLines: null,
};
const h5Proptypes = {
  /**
   * The prop "style" overrides the custom styles set in the component.
   */
  style: object,
  /**
   * The prop "color" defines the color of the text.
   */
  color: bool,
  /**
   * The prop "numberOfLines" receives a number which limits the lines shown
   * (an ellipsis will show at the end when truncated)
   */
  numberOfLines: number,
  /**
   * The prop "children" defines the content that will live inside the component.
   */
  children: string.isRequired,
};
const h5DefaultProps = {
  style: {},
  color: false,
  numberOfLines: null,
};
const textProptypes = {
  /**
   * The prop "style" overrides the custom styles set in the component.
   */
  style: object,
  /**
   * The prop "numberOfLines" receives a number which limits the lines shown
   * (an ellipsis will show at the end when truncated)
   */
  numberOfLines: number,
  /**
   * The prop "children" defines the content that will live inside the component.
   */
  children: string.isRequired,
};
const textDefaultProps = {
  style: {},
  numberOfLines: null,
};

/**
 * H1 component renders a RN <Text> component with custom styles.
 */
const H1 = props => (
  <Text
    numberOfLines={props.numberOfLines}
    style={props.color ? { ...styles.h1Color, ...props.style } : { ...styles.h1, ...props.style }}
  >
    {props.children}
  </Text>
);
/**
 * H2 component renders a RN <Text> component with custom styles.
 */
const H2 = props => (
  <Text
    numberOfLines={props.numberOfLines}
    style={props.color ? { ...styles.h2Color, ...props.style } : { ...styles.h2, ...props.style }}
  >
    {props.children}
  </Text>
);
/**
 * H3 component renders a RN <Text> component with custom styles.
 */
const H3 = props => (
  <Text
    numberOfLines={props.numberOfLines}
    style={props.color ? { ...styles.h3Color, ...props.style } : { ...styles.h3, ...props.style }}
  >
    {props.children}
  </Text>
);
/**
 * H4 component renders a RN <Text> component with custom styles.
 */
const H4 = props => (
  <Text
    numberOfLines={props.numberOfLines}
    style={props.color ? { ...styles.h4Color, ...props.style } : { ...styles.h4, ...props.style }}
  >
    {props.children}
  </Text>
);
/**
 * H5 component renders a RN <Text> component with custom styles.
 */
const H5 = props => (
  <Text
    numberOfLines={props.numberOfLines}
    style={props.color ? { ...styles.h5Color, ...props.style } : { ...styles.h5, ...props.style }}
  >
    {props.children}
  </Text>
);
/**
 * Text component renders a RN <Text> component with custom styles.
 */
const TextTheme = props => (
  <Text numberOfLines={props.numberOfLines} style={{ ...styles.text, ...props.style }}>
    {props.children}
  </Text>
);

H1.propTypes = h1Proptypes;
H1.defaultProps = h1DefaultProps;
H2.propTypes = h2Proptypes;
H2.defaultProps = h2DefaultProps;
H3.propTypes = h3Proptypes;
H3.defaultProps = h3DefaultProps;
H4.propTypes = h4Proptypes;
H4.defaultProps = h4DefaultProps;
H5.propTypes = h5Proptypes;
H5.defaultProps = h5DefaultProps;
TextTheme.propTypes = textProptypes;
TextTheme.defaultProps = textDefaultProps;

export { H1, H2, H3, H4, H5, TextTheme as Text };
