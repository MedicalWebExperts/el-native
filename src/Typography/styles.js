import colors from '../Theme/colors';

export default {
  h1: {
    fontWeight: '600',
    fontSize: 20,
    color: colors.darkTextColor,
  },
  h1Color: {
    fontWeight: '600',
    fontSize: 20,
    color: colors.blueColor,
  },
  h2: {
    fontWeight: '600',
    fontSize: 18,
    color: colors.textColor,
  },
  h2Color: {
    fontWeight: '600',
    fontSize: 18,
    color: colors.blueColor,
  },
  h3: {
    fontWeight: '500',
    fontSize: 16,
    color: colors.textColor,
  },
  h3Color: {
    fontWeight: '500',
    fontSize: 16,
    color: colors.blueColor,
  },
  h4: {
    fontWeight: '500',
    fontSize: 14,
    color: colors.textColor,
  },
  h4Color: {
    fontWeight: '500',
    fontSize: 14,
    color: colors.blueColor,
  },
  h5: {
    fontWeight: '600',
    fontSize: 12,
    color: colors.textColor,
  },
  h5Color: {
    fontWeight: '600',
    fontSize: 12,
    color: colors.blueColor,
  },
  text: {
    fontWeight: '400',
    fontSize: 14,
    color: colors.textColor,
  },
};
