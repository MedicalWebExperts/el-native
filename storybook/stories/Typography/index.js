import React from 'react';
import { storiesOf } from '@storybook/react-native';

import CenterView from '../CenterView';
import { H1, H2, H3, H4, H5, Text } from '../../../src';

storiesOf('Map', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('H1', () => <H1>{"I'm the migthy H1."}</H1>)
  .add('H1 Color', () => <H1 color>{"I'm the migthy H1, dressed in colors."}</H1>)
  .add('H2', () => <H2>{"I'm H2, nothing wrong with being second."}</H2>)
  .add('H2 Color', () => <H2 color>{"I'm H2. With colors!"}</H2>)
  .add('H3', () => <H3>{"I'm H3, how are you?"}</H3>)
  .add('H3 Color', () => <H3 color>{"I'm H3 in colors."}</H3>)
  .add('H4', () => <H4>{'H4 here, hi!'}</H4>)
  .add('H4 Color', () => <H4 color>{'H4 in colors.'}</H4>)
  .add('H5', () => <H5>{"I'm H5, and I'm glad I am"}</H5>)
  .add('H5 Color', () => <H5 color>{"I'm H5 coloured."}</H5>)
  .add('Text', () => <Text>{"I'm just and old plain Text."}</Text>);
