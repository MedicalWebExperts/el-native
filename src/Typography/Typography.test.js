/* global describe it expect jest:true */
import React from 'react';
import renderer from 'react-test-renderer';

import { H1, H2, H3, H4, H5, Text } from './Typography';

const props = {
  style: {},
  color: false,
  children: 'Hi there!',
};

const textProps = {
  style: {},
  children: 'Hi there!',
};

describe('H1 Snapshots', () => {
  it('should render an H1', () => {
    const component = renderer.create(<H1 {...props} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render an H1 with color', () => {
    const component = renderer.create(<H1 {...props} color />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should apply custom styles to H1', () => {
    const component = renderer.create(
      <H1
        {...props}
        style={{
          color: 'green',
        }}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('H2 Snapshots', () => {
  it('should render an H2', () => {
    const component = renderer.create(<H2 {...props} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render an H2 with color', () => {
    const component = renderer.create(<H2 {...props} color />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should apply custom styles to H2', () => {
    const component = renderer.create(
      <H2
        {...props}
        style={{
          color: 'green',
        }}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('H3 Snapshots', () => {
  it('should render an H3', () => {
    const component = renderer.create(<H3 {...props} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render an H3 with color', () => {
    const component = renderer.create(<H3 {...props} color />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should apply custom styles to H3', () => {
    const component = renderer.create(
      <H3
        {...props}
        style={{
          color: 'green',
        }}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('H4 Snapshots', () => {
  it('should render an H4', () => {
    const component = renderer.create(<H4 {...props} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render an H4 with color', () => {
    const component = renderer.create(<H4 {...props} color />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should apply custom styles to H4', () => {
    const component = renderer.create(
      <H4
        {...props}
        style={{
          color: 'green',
        }}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('H5 Snapshots', () => {
  it('should render an H5', () => {
    const component = renderer.create(<H5 {...props} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render an H5 with color', () => {
    const component = renderer.create(<H5 {...props} color />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should apply custom styles to H5', () => {
    const component = renderer.create(
      <H5
        {...props}
        style={{
          color: 'green',
        }}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Text Snapshots', () => {
  it('should render an Text', () => {
    const component = renderer.create(<Text {...textProps} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render an Text with color', () => {
    const component = renderer.create(<Text {...textProps} color />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should apply custom styles to Text', () => {
    const component = renderer.create(
      <Text
        {...textProps}
        style={{
          color: 'green',
        }}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
