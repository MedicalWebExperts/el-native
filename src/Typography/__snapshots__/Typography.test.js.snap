// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 Snapshots should apply custom styles to H1 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "green",
      "fontSize": 48,
      "fontWeight": "600",
    }
  }
>
  Hi there!
</Text>
`;

exports[`H1 Snapshots should render an H1 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "#212121",
      "fontSize": 48,
      "fontWeight": "600",
    }
  }
>
  Hi there!
</Text>
`;

exports[`H1 Snapshots should render an H1 with color 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "#2196F3",
      "fontSize": 48,
      "fontWeight": "600",
    }
  }
>
  Hi there!
</Text>
`;

exports[`H2 Snapshots should apply custom styles to H2 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "green",
      "fontSize": 36,
      "fontWeight": "600",
    }
  }
>
  Hi there!
</Text>
`;

exports[`H2 Snapshots should render an H2 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "#212121",
      "fontSize": 36,
      "fontWeight": "600",
    }
  }
>
  Hi there!
</Text>
`;

exports[`H2 Snapshots should render an H2 with color 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "#2196F3",
      "fontSize": 36,
      "fontWeight": "600",
    }
  }
>
  Hi there!
</Text>
`;

exports[`H3 Snapshots should apply custom styles to H3 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "green",
      "fontSize": 24,
      "fontWeight": "500",
    }
  }
>
  Hi there!
</Text>
`;

exports[`H3 Snapshots should render an H3 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "#212121",
      "fontSize": 24,
      "fontWeight": "500",
    }
  }
>
  Hi there!
</Text>
`;

exports[`H3 Snapshots should render an H3 with color 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "#2196F3",
      "fontSize": 24,
      "fontWeight": "500",
    }
  }
>
  Hi there!
</Text>
`;

exports[`H4 Snapshots should apply custom styles to H4 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "green",
      "fontSize": 20,
      "fontWeight": "500",
    }
  }
>
  Hi there!
</Text>
`;

exports[`H4 Snapshots should render an H4 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "#212121",
      "fontSize": 20,
      "fontWeight": "500",
    }
  }
>
  Hi there!
</Text>
`;

exports[`H4 Snapshots should render an H4 with color 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "#2196F3",
      "fontSize": 20,
      "fontWeight": "500",
    }
  }
>
  Hi there!
</Text>
`;

exports[`H5 Snapshots should apply custom styles to H5 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "green",
      "fontSize": 16,
      "fontWeight": "600",
    }
  }
>
  Hi there!
</Text>
`;

exports[`H5 Snapshots should render an H5 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "#212121",
      "fontSize": 16,
      "fontWeight": "600",
    }
  }
>
  Hi there!
</Text>
`;

exports[`H5 Snapshots should render an H5 with color 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "#2196F3",
      "fontSize": 16,
      "fontWeight": "600",
    }
  }
>
  Hi there!
</Text>
`;

exports[`Text Snapshots should apply custom styles to Text 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "green",
      "fontSize": 14,
      "fontWeight": "400",
    }
  }
>
  Hi there!
</Text>
`;

exports[`Text Snapshots should render an Text 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "#212121",
      "fontSize": 14,
      "fontWeight": "400",
    }
  }
>
  Hi there!
</Text>
`;

exports[`Text Snapshots should render an Text with color 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={null}
  style={
    Object {
      "color": "#212121",
      "fontSize": 14,
      "fontWeight": "400",
    }
  }
>
  Hi there!
</Text>
`;
